assets/in

```diff
@@ -16,26 +18,49 @@ fi
 
 # parse incoming config data
 payload=$(cat)
-bucket=$(echo "${payload}" | jq -r '.source.bucket')
+
+bucket="$(echo "${payload}" | jq -r '.source.bucket')"
+if [ -z "${bucket}" ]; then
+  echo "No bucket supplied"
+  exit 1
+fi
+
 endpoint_url="$(echo "${payload}" | jq -r '.source.endpoint // ""')"
 region="$(echo "${payload}" | jq -r '.source.region // "us-east-1"')"
 use_v4="$(echo "${payload}" | jq -r '.source.use_v4 // ""')"
-path=$(echo "${payload}" | jq -r '.source.path // ""')
-suffix=$(echo "${payload}" | jq -r '.source.suffix // ""')
+path="$(echo "${payload}" | jq -r '.source.path // ""')"
+if [ "${path}" = '/' ]; then
+  path=""
+fi
+suffix="$(echo "${payload}" | jq -r '.source.suffix // ""')"
 encryption_key="$(echo "${payload}" | jq -r '.source.encryption_key // ""')"
 skip_ssl_verification="$(echo "$payload" | jq -r '.source.skip_ssl_verification // ""')"
 ca_bundle="$(echo "$payload" | jq -r '.source.ca_bundle // ""')"
+minio_identifier="minio" # used to check if endpoint_url contains 'minio'
```

Comment:
Does it have to be `minio`? This is non FIL specific resource and we can't assume this.